include/aml.h

```diff
@@ -929,50 +924,54 @@ struct aml_dma_ops {
 	/**
 	 * Initiate a data movement, from a source pointer to a destination
 	 * pointer, and output a request handler for managing the transfer.
-	 * @param dma: dma_implementation internal data.
-	 * @param req[out]: the request handle to manage termination
+	 * @param[in, out] dma: dma_implementation internal data.
+	 * @param[in, out] req: the request handle to manage termination
 	 *        of the movement.
-	 * @param dest: layout describing the destination.
-	 * @param src: layout describing the source.
+	 * @param[out] dest: layout describing the destination.
+	 * @param[in] src: layout describing the source.
 	 * @return an AML error code.
 	 **/
 	int (*create_request)(struct aml_dma_data *dma,
-			      struct aml_dma_request **req,
-			      struct aml_layout *dest,
-			      struct aml_layout *src,
-			      aml_dma_operator op, void *op_arg);
+	                      struct aml_dma_request **req,
+	                      struct aml_layout *dest,
+	                      struct aml_layout *src,
+	                      aml_dma_operator op,
+	                      void *op_arg);
 
 	/**
 	 * Destroy the request handle. If the data movement is still ongoing,
 	 * then cancel it.
 	 *
-	 * @param dma: dma_implementation internal data.
-	 * @param req: the request handle to manage termination of the movement.
+	 * @param[in] dma: dma_implementation internal data.
+	 * @param[in] req: the request handle to manage termination of the
```

Comment:
```suggestion
	 * @param[inout] req: the request handle to manage termination of the
```
If I remember correctly, this one gets set to NULL on exit.